src/@next/Select/Select.tsx

```diff
@@ -43,24 +42,54 @@ export interface SelectProps {
 
 export const Select = ({
   allowMultiple = false,
-  disabled,
-  hasError,
+  disabled = false,
+  hasError = false,
   helpText,
   label,
   onClose,
   onRemoveTag,
   onSelect,
-  options,
+  options = [],
   placeholder,
   listHeight,
+  searchable = false,
   searchableProps,
-  scrollable,
+  scrollable = false,
   sections,
   selectedValues,
   width,
 }: SelectProps) => {
   const [popoverActive, setPopoverActive] = useState(false);
   const [optionListHeight, setOptionListHeight] = useState('');
+  const [menuOptions, setMenuOptions] = useState(options);
+  const [inputValue, setInputValue] = useState(
+    searchableProps?.inputValue || ''
+  );
+
+  const [searchableSelectState, setSearchableSelectState] =
+    useState<SearchableSelectState>({
+      showSelected: false,
+      showInput: true,
+      showPlaceholder: true,
+    });
+
+  const updateSearchableSelectState = (newState: SearchableSelectState) => {
+    setSearchableSelectState(newState);
+  };
+
+  const updateInputValue = useCallback(
```

Comment:
can we confirm if we still need the useCallback? Same for others

Reply:
thanks for the catch. this should be removeable as well